app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_of_people = [Person(person_data["name"], person_data["age"])
+                      for person_data in people]
+
+    for person_data in people:
+        name = person_data["name"]
+        person = Person.people[name]
+
+        if "wife" in person_data:
+            wife_name = person_data.get("wife")
+            if wife_name:
+                spouse = Person.people[wife_name]
+                person.wife = spouse
+
+        if "husband" in person_data:
+            husband_name = person_data.get("husband")
```

Comment:
If you call `if person_data.get("husband"):` it will check` if "husband" in person_data:` And `if husband_name:`

So you will have only 1 check instead of 2